include/misc/util.h

```diff
@@ -125,6 +125,14 @@ static inline s64_t arithmetic_shift_right(s64_t value, u8_t shift)
 #endif
 #endif
 
+#ifndef BIT_U8
+#if defined(_ASMLANGUAGE)
+#define BIT_U8(n)  (1 << (n))
+#else
+#define BIT_U8(n)  ((u8_t)1 << (n))
+#endif
+#endif
+
```

Comment:
do we really need these macros ? As far as I could see they are being used only in three places, wouldn't be better just cast the return there ? I'm totally in keep API as slim as possible.

Reply:
Actually we might have to start using this macro in other places aswell. As of now it is used in 3 places.
I didn't get to the other places just yet.

Reply:
Yeah, but what happens if you need something for uint16_t ? A new macro ?
I guess we can do something like:

```
#ifndef BIT
#if defined(_ASMLANGUAGE)
#define BIT(n, ...)  (1 << (n))
#else
#define BIT(n, ...)  (__VA_ARGS__ 1 << (n))
#endif
#endif


BIT(3)
BIT(3, (u8_t))
```

What do you think ?

Reply:
I also don't see why we need these new macros, or any new params to the exisiting macro. If there're concern about signedness, the existing BIT() definition should be modified to be `1u << (n)`. Any casts should be done to the outside expression.